### PR TITLE
fix: ReportTable not updating status dynamically

### PR DESCRIPTION
**Pull request by andreas-unleash**
Fixes a bug with the UI where featureType comparison was using name instead of id

### CODE REVIEW

File: frontend/src/component/project/Project/ProjectHealth/ReportTable/ReportTable.tsx [hunk 1]
```diff
@@ -59,21 +59,20 @@ export const ReportTable = ({ projectId, features }: IReportTableProps) => {
...
+            name: report.name,
+            type: report.type,
+            stale: report.stale,
+            environments: report.environments,
+            status: formatStatus(report, featureTypes),
+            lastSeenAt: report.lastSeenAt,
+            createdAt: report.createdAt,
+            expiredAt: formatExpiredAt(report, featureTypes),
+        }));
+    }, [projectId, features, featureTypes]);
```
Comment (Tymek) [inline frontend/src/component/project/Project/ProjectHealth/ReportTable/ReportTable.tsx:75]:
```suggestion
    const data: IReportTableRow[] = useMemo<IReportTableRow[]>(
        () =>
            features.map(report => ({
                project: projectId,
                name: report.name,
                type: report.type,
                stale: report.stale,
                environments: report.environments,
                status: formatStatus(report, featureTypes),
                lastSeenAt: report.lastSeenAt,
                createdAt: report.createdAt,
                expiredAt: formatExpiredAt(report, featureTypes),
            })),
        [projectId, features]
    );
```

File: frontend/src/component/project/Project/ProjectHealth/ReportTable/ReportTable.tsx [hunk 2]
```diff
@@ -145,7 +144,7 @@ export const ReportTable = ({ projectId, features }: IReportTableProps) => {
                 maxWidth: 120,
             },
         ],
-        [showEnvironmentLastSeen]
+        [showEnvironmentLastSeen, JSON.stringify(data)]
```
Comment (gastonfournier) [inline frontend/src/component/project/Project/ProjectHealth/ReportTable/ReportTable.tsx:147]: Is JSON.stringify(data) really needed? I guess this is to make it change if anything on the response changes, but it feels excessive

Comment (Tymek) [inline frontend/src/component/project/Project/ProjectHealth/ReportTable/ReportTable.tsx:147]: Is this needed?

Comment (andreas-unleash) [reply frontend/src/component/project/Project/ProjectHealth/ReportTable/ReportTable.tsx]: removed

File: frontend/src/component/project/Project/ProjectHealth/ReportTable/ReportTable.tsx (hunk 1 above)
Comment (andreas-unleash) [reply frontend/src/component/project/Project/ProjectHealth/ReportTable/ReportTable.tsx]: removed

File: frontend/src/component/project/Project/ProjectHealth/ReportTable/ReportTable.tsx (hunk 2 above)
Comment (andreas-unleash) [reply frontend/src/component/project/Project/ProjectHealth/ReportTable/ReportTable.tsx]: removed

**Review by gastonfournier — APPROVED** [review]
LGTM

**Review by Tymek — APPROVED** [review]

**Pull request merged**